src/core/modes/walk-mode.ts

```diff
@@ -159,4 +171,25 @@ export class WalkMode implements Mode {
 
     this.decorationTimer = setInterval(updateComboTimerDecoration, 50);
   }
+
+  private getLocation(firstVisibleRange: vscode.Range) {
+    const location = vscode.workspace.getConfiguration('junbae-mode').get('location');
+
+    let visibleStartLine = firstVisibleRange.start;
+    let visibleEndLine = firstVisibleRange.start;
+    const junbaeLocation = { top: '5%', 'font-size': '50px', bottom: 'none' };
+    const timerLocation = { top: '5%', bottom: 'none' };
+
+    if (location === 'bottom') {
+      visibleStartLine = firstVisibleRange.end;
+      visibleEndLine = firstVisibleRange.end;
+      junbaeLocation.top = 'none';
+      junbaeLocation.bottom = '10px';
+      junbaeLocation['font-size'] = '100px';
+      timerLocation.top = 'none';
+      timerLocation.bottom = '70px';
+    }
+
+    return { junbaeLocation, timerLocation, visibleStartLine, visibleEndLine };
```

Comment:
`visibleStartLine`, `visibleEndLine`

are they location config?
if it is, you can return `range`